plugins/inmemory-records-storage/src/main/java/com/github/bsideup/liiklus/records/inmemory/InMemoryRecordsStorage.java

```diff
@@ -157,12 +181,45 @@ public int getPartition() {
         };
     }
 
+    private static Envelope toEnvelope(String topic, StoredTopic.Partition.Record record) {
+        ByteBuffer keyBuffer = record.getKey();
+        Object valueBuffer = record.getValue();
+        Map<String, String> headers = record.getHeaders();
+
+        var key = keyBuffer != null ? keyBuffer.asReadOnlyBuffer() : null;
+        var specVersion = headers.get("ce-specversion");
+        if (specVersion == null) {
+            return new Envelope(
+                    topic,
+                    key,
+                    ((ByteBuffer) valueBuffer).asReadOnlyBuffer()
+            );
+        }
+        switch (specVersion) {
+            case "1.0":
+                return new Envelope(
+                        topic,
+
+                        key,
+                        it -> (ByteBuffer) it,
+
+                        EVENT_UNMARSHALLER
+                                .withHeaders(() -> (Map) headers)
+                                .withPayload(() -> (String) valueBuffer)
+                                .unmarshal(),
+                        it -> ByteBuffer.wrap(Json.binaryEncode(it)).asReadOnlyBuffer()
```

Comment:
I have yet to fully review what this PR exactly does, but the usage of JSON seems wrong. In memory should just store events as object references, shouldn't it?
(and we don't want to use JSON anywhere, in-memory or kafka, as a matter of fact)

Reply:
this `Function` converts the in-memory representation to a `ByteBuffer` when pre/post-processor is unable to handle the in-memory format, mostly due to legacy reasons.

Eventually, the processors API will be changed to use `CloudEvent` as the type, not `Object`, and the functions will go away.

In other words, we use `CloudEvent` as the in-memory type and we fallback to the JSON representation when bytes are requested (either `BINARY` consumer format or pre/post-processors)

Reply:
D'oh! Of course 1 second after replying I got what you mean 😂 

in-memory forces the serialization to avoid hard references and non-serializable objects that otherwise would fail with kafka/pulsar/any other real storage. JSON was selected as the easiest